bodhi/server/static/js/update_form.js

```diff
@@ -298,16 +303,93 @@ $(document).ready(function() {
     // field or the 'bugs' field, those things get added to the list of
     // possibilities.
     var add_bugs = function() {
+        $("#bugs-checkboxes").prepend("<img class='spinner' src='static/img/spinner.gif'>\n");
         var value = $("#bugs-adder input").val().trim();
+        var count_done = 0;
+        var count_todo = 0;
         $.each(value.split(","), function(i, intermediary) {
-            $.each(intermediary.trim().split(" "), function(j, item) {
+            bug_list = intermediary.trim().split(" ");
+            count_todo += bug_list.length;
+            $.each(bug_list, function(j, item) {
                 item = item.trim()
                 if (item[0] == '#') { item = item.substring(1); }
-                add_bug_checkbox(item, '', true, true);
+                // Prevent empty item e.g. double space in input
+                if (item == '') {
+                    count_done += 1;
+                    return;
+                }
+                $.ajax({
+                    url: "https://bugzilla.redhat.com/jsonrpc.cgi?method=Bug.get&params=%5B%7B%22ids%22:" + item + "%7D%5D",
+                    timeout: 10000,
+                    dataType: 'jsonp',
+                    success: function(data) {
+                        // Received error response
+                        if (data.error) {
+                            // Error 102 is due to bug being private
+                            // Bodhi can't handle private bugs, we should avoid
+                            // attaching them to updates
+                            if (data.error.code == 102) {
+                                messenger.post({
+                                    message: 'Bodhi can\'t manage private bugs! (#' + item + ')',
+                                    type: 'error',
+                                });
+                            }
+                            else {
+                                messenger.post({
+                                    message: data.error.message,
+                                    type: 'error',
+                                });
+                            }
+                            return;
+                        }
+                        // There should be only one
+                        if (data.result.bugs.length != 1) {
+                            messenger.post({
+                                message: 'Received bad response for bug #' + item,
+                                type: 'error',
+                            });
+                            return;
+                        }
+                        // Check Bug product
+                        var product = data.result.bugs[0].product;
+                        if (product !== "Fedora" && product !== "Fedora EPEL" && product !== "Fedora Modules") {
```

Comment:
There are other Bodhi deployments in the wild that do not have products named Fedora or EPEL. We should not hard code product names here.

Reply:
Well, the ajax in the script already fetch the bugs list of the selected package from `apps.fedoraproject.org/packages/fcomm_connector/bugzilla/`, which returns only Fedora, EPEL and (recently applied patch) Fedora Modules. So, in some way, we are already hardcoding allowed products.

Is setting a list of allowed products at the top of the file acceptable to you, or should I remove (or simply comment out) the product check?

Reply:
Yeah Bodhi does have some hardcoded behavior - I've been slowly trying to remove it as I go over time. It's one of the things that makes it hard for the other deployments, or for new deployments (I've talked to a few people who tried to use it and hit these hardcoded things.)

One way to solve this problem is to make the list of product names to check (and the URL to bugs, for that matter) a setting in ```bodhi.server.config```, so that other deployments can configure it as they see fit.

Reply:
I've changed the product check to use config['bz_products'] list already defined in settings. However, I added the defaults into the server/config.py file rather than rely on production.ini, mainly because this way it will also work in local development environment without adding the same setting to development.ini.